cassiopeia/plotting/local.py

```diff
@@ -0,0 +1,830 @@
+    if len(set.union(*list(descendants.values()))) != sum(
+        len(d) for d in descendants.values()
+    ):
+        warnings.warn(
+            "Some clades specified with `clade_colors` are overlapping. "
+            "Colors may be overridden.",
+            PlottingWarning,
+        )
+
+    # Color by largest clade first
+    node_colors = {}
+    branch_colors = {}
+    for node in sorted(
+        descendants, key=lambda x: len(descendants[x]), reverse=True
+    ):
+        color = clade_colors[node]
+        for n1, n2 in tree.depth_first_traverse_edges(node):
+            node_colors[n1] = node_colors[n2] = color
+            branch_colors[(n1, n2)] = color
+    return node_colors, branch_colors
+
+
+def place_tree_and_annotations(
+    tree: CassiopeiaTree,
+    depth_key: Optional[str] = None,
+    meta_data: Optional[List[str]] = None,
+    allele_table: Optional[pd.DataFrame] = None,
+    indel_colors: Optional[pd.DataFrame] = None,
+    indel_priors: Optional[pd.DataFrame] = None,
+    orient: Union[Literal["up", "down", "left", "right"], float] = 90.0,
+    extend_branches: bool = True,
+    angled_branches: bool = True,
+    add_root: bool = False,
+    colorstrip_width: Optional[float] = None,
+    colorstrip_spacing: Optional[float] = None,
+    clade_colors: Optional[Dict[str, Tuple[float, float, float]]] = None,
+    continuous_cmap: Union[str, mpl.colors.Colormap] = "viridis",
+    categorical_cmap: Union[str, mpl.colors.Colormap] = "tab10",
+    random_state: Optional[np.random.RandomState] = None,
+) -> Tuple[Dict, Dict, Dict, Dict, List]:
+    """Helper function to place the tree and all requested annotations.
+
+    Args:
+        tree: The CassiopeiaTree to plot.
+        depth_key: The node attribute to use as the depth of the nodes. If
+            not provided, the distances from the root is used by calling
+            `tree.get_distances`.
+        meta_data: Meta data to plot alongside the tree, which must be columns
+            in the CassiopeiaTree.cell_meta variable.
+        allele_table: Alleletable to plot alongside the tree.
+        indel_colors: Color mapping to use for plotting the alleles for each
+            cell. Only necessary if `allele_table` is specified.
+        indel_priors: Prior probabilities for each indel. Only useful if an
+            allele table is to be plotted and `indel_colors` is None.
+        orient: The orientation of the tree. Valid arguments are `left`, `right`,
+            `up`, `down` to display a rectangular plot (indicating the direction
+            of going from root -> leaves) or any number, in which case the
+            tree is placed in polar coordinates with the provided number used
+            as an angle offset. Defaults to 90.
+        extend_branches: Extend branch lengths such that the distance from the
+            root to every node is the same. If `depth_key` is also provided, then
+            only the leaf branches are extended to the deepest leaf.
```

Comment:
This is a somewhat arbitrary decision on my part so that the `depth_key` and `extend_branches` arguments work nicely with each other. Also, I previously saw Sebastian extending only the leaves to the deepest leaf in his branch length estimation stuff, so that's why I decided to include this.

Reply:
I agree - this is probably the best way to do this.